### PR TITLE
Bluetooth: Mesh: Fixed delete scene pages.

### PR DESCRIPTION
**Pull request by Balaklaka**
When the data was stored in pages the counter that kept the number
of pages stored was not updated correctly, so when a scene was deleted
there was pages left that was not deleted from the file system.

Signed-off-by: Ingar Kulbrandstad <ingar.kulbrandstad@nordicsemi.no>

### CODE REVIEW

File: subsys/bluetooth/mesh/scene_srv.c [hunk 1]
```diff
@@ -20,6 +20,13 @@
 /* Account for company ID in data: */
 #define VND_MODEL_SCENE_DATA_OVERHEAD sizeof(uint16_t)
 
+#define SCENE_PAGES_COUNT_SET(_vnd, _srv, _page)                               \
+	if (_vnd) {                                                            \
+		(_srv)->vndpages = MAX((_page) + 1, (_srv)->vndpages);         \
+	} else {                                                               \
+		(_srv)->sigpages = MAX((_page) + 1, (_srv)->sigpages);         \
+	}
```
[the remark this answers is not in the document]
Comment (Balaklaka) [reply subsys/bluetooth/mesh/scene_srv.c]: static inline function created.

File: subsys/bluetooth/mesh/scene_srv.c [hunk 2]
```diff
@@ -386,7 +388,7 @@ static enum bt_mesh_scene_status scene_store(struct bt_mesh_scene_srv *srv,
 
 	page = 0;
 
-	SYS_SLIST_FOR_EACH_CONTAINER(&srv->vnd, entry, n) {
+	SYS_SLIST_FOR_EACH_CONTAINER (&srv->vnd, entry, n) {
```
[the remark this answers is not in the document]
Comment (Balaklaka) [reply subsys/bluetooth/mesh/scene_srv.c]: Fixed

File: subsys/bluetooth/mesh/scene_srv.c [hunk 3]
```diff
@@ -33,6 +34,16 @@ static char *scene_path(char *buf, uint16_t scene, bool vnd, uint8_t page)
 	return buf;
 }
 
+static inline void pages_count(bool vnd, uint8_t page, 
+			       struct bt_mesh_scene_srv *srv)
```
Comment (trond-snekvik) [inline subsys/bluetooth/mesh/scene_srv.c:38]: `update_page_count` or something? There's no way to tell what the purpose of this is based on the name. Also, the model pointer is always the first argument in model functions.

Comment (Balaklaka) [reply subsys/bluetooth/mesh/scene_srv.c]: Renamed.

**Review by trond-snekvik — APPROVED** [review]

**Review by ludvigsj — APPROVED** [review]

**Pull request merged**